lib/rex/post/mssql/ui/console/command_dispatcher/client.rb

```diff
@@ -0,0 +1,148 @@
+          def initialize(console)
+            super
+
+            @db_search_results = []
+          end
+
+          #
+          # List of supported commands.
+          #
+          # @return [Hash{String->String}]
+          def commands
+            cmds = {
+              'query'   => 'Run a raw SQL query',
+              'shell'   => 'Enter a raw shell where SQL queries can be executed',
+            }
+
+            reqs = {}
+
+            filter_commands(cmds, reqs)
+          end
+
+          # @return [String]
+          def name
+            'MSSQL Client'
+          end
+
+          # @param [Object] args
+          # @return [FalseClass, TrueClass]
+          def help_args?(args)
+            return false unless args.instance_of?(::Array)
+
+            args.include?('-h') || args.include?('--help')
+          end
+
+          # @return [Object]
+          def cmd_shell_help
+            print_line 'Usage: shell'
+            print_line
+            print_line 'Go into a raw SQL shell where SQL queries can be executed.'
+            print_line 'To exit, type `exit`, `quit`, `end` or `stop`.'
+            print_line
+          end
+
+          # @param [Array] args
+          # @return [Object]
+          def cmd_shell(*args)
+            cmd_shell_help && return if help_args?(args)
+
+            prompt_proc_before = ::Reline.prompt_proc
+
+            ::Reline.prompt_proc = proc { |line_buffer| line_buffer.each_with_index.map { |_line, i| i > 0 ? 'SQL *> ' : 'SQL >> ' } }
+
+            stop_words = %w[stop s exit e end quit q].freeze
+
+            finished = false
+            until finished
+              begin
+                raw_query = ::Reline.readmultiline('SQL >> ', use_history = true) do |multiline_input|
+                  finished = stop_words.include?(multiline_input.split.last)
+                  !multiline_input.split.last.end_with?('\\')
+                end
+              rescue ::Interrupt
+                finished = true
+              ensure
+                ::Reline.prompt_proc = prompt_proc_before
+              end
+
+              if finished
+                print_status 'Exiting Shell mode.'
+                return
+              end
+
+              formatted_query = raw_query.split.map { |word| word.chomp('\\') }.reject(&:empty?).compact.join(' ')
+
+              print_status "Running SQL Command: '#{formatted_query}'"
+              cmd_query(formatted_query)
+            end
+          end
+
+          # @return [Object]
+          def cmd_query_help
+            print_line 'Usage: query'
+            print_line
+            print_line 'Run a raw SQL query on the target.'
+            print_line 'Examples:'
+            print_line
+            print_line '    query SHOW DATABASES;'
+            print_line '    query USE information_schema;'
+            print_line '    query SELECT * FROM SQL_FUNCTIONS;'
+            print_line '    query SELECT version();'
+            print_line
```

Comment:
This is Postgres stuff, should replace with MSSQL commands that do +- same things from [this list](https://book.hacktricks.xyz/network-services-pentesting/pentesting-mssql-microsoft-sql-server#common-enumeration)